keyboards/kprepublic/bm60hsrgb_ec/rev2/rev2.c

```diff
@@ -0,0 +1,200 @@
+#    if WS2812_LED_TOTAL > 0
+        ,{ 28, 40}, { 62, 40}, { 96, 40}, {130, 40}, {164, 40}, {198, 40}
+#    endif
+}, {
+    // Esc, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, -, =, Backspace
+    4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 1,
+    // Tab, Q, W, E, R, T, Y, U, I, O, P, [, ],
+    1, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+    // Capslock, A, S, D, F, G, H, A, K, L, ;, ', Enter
+    9, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 1,
+    // LShift, Z, X, C, V, B, N, M, ,, ., /, Shift, Up, ?
+    1, 4, 4, 4, 4, 4, 4, 4, 4, 4, 1, 1, 4,
+    // Ctrl, GUI, Alt, Space, RAlt, FN, Left, Down, Right
+    1, 1, 1, 4, 1, 1, 1, 1, 1
+#    if WS2812_LED_TOTAL > 0
+        ,2, 2, 2, 2, 2, 2
+#    endif
+    }
+};
+
+__attribute__ ((weak))
+void rgb_matrix_indicators_user(void) {
+    if (host_keyboard_led_state().caps_lock) {
+        rgb_matrix_set_color(28, 0xFF, 0xFF, 0xFF);
+    }
+}
+
+
+// clang-format on
+
+// ==========================================================================
+// Custom RGB Matrix driver that combines IS31FL3733 and WS2812
+// ==========================================================================
+
+#    if WS2812_LED_TOTAL > 0
+LED_TYPE rgb_matrix_ws2812_array[WS2812_LED_TOTAL];
+#    endif
+
+static void rgb_matrix_driver_init(void) {
+    i2c_init();
+    IS31FL3733_init(DRIVER_ADDR_1, 0);
+    for (uint8_t index = 0; index < ISSI_LED_TOTAL; index++) {
+        bool enabled = true;
+        IS31FL3733_set_led_control_register(index, enabled, enabled, enabled);
+    }
+    IS31FL3733_update_led_control_registers(DRIVER_ADDR_1, 0);
+}
+
+static void rgb_matrix_driver_flush(void) {
+    IS31FL3733_update_pwm_buffers(DRIVER_ADDR_1, 0);
+#    if WS2812_LED_TOTAL > 0
+    ws2812_setleds(rgb_matrix_ws2812_array, WS2812_LED_TOTAL);
+#    endif
+}
+
+static void rgb_matrix_driver_set_color(int index, uint8_t red, uint8_t green, uint8_t blue) {
+    if (index < ISSI_LED_TOTAL) {
+        IS31FL3733_set_color(index, red, green, blue);
+    } else {
+#    if WS2812_LED_TOTAL > 0
+        rgb_matrix_ws2812_array[index - ISSI_LED_TOTAL].r = red;
+        rgb_matrix_ws2812_array[index - ISSI_LED_TOTAL].g = green;
+        rgb_matrix_ws2812_array[index - ISSI_LED_TOTAL].b = blue;
+#    endif
+    }
+}
+
+static void rgb_matrix_driver_set_color_all(uint8_t red, uint8_t green, uint8_t blue) {
+    IS31FL3733_set_color_all(red, green, blue);
+#    if WS2812_LED_TOTAL > 0
+    for (uint8_t i = 0; i < WS2812_LED_TOTAL; i++) {
+        rgb_matrix_ws2812_array[i].r = red;
+        rgb_matrix_ws2812_array[i].g = green;
+        rgb_matrix_ws2812_array[i].b = blue;
+    }
+#    endif
+}
+
+// clang-format off
+const rgb_matrix_driver_t rgb_matrix_driver = {
+    .init          = rgb_matrix_driver_init,
+    .flush         = rgb_matrix_driver_flush,
+    .set_color     = rgb_matrix_driver_set_color,
+    .set_color_all = rgb_matrix_driver_set_color_all,
+};
+// clang-format on
+
+#endif /* RGB_MATRIX_ENABLE */
```

Comment:
```suggestion
#endif /* RGB_MATRIX_ENABLE */

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) {
        return false;
    }
    if (layer_state_is(1)) {
	      if (clockwise) {
	          tap_code(KC_VOLU);
	      } else {
	          tap_code(KC_VOLD);
	      }
	} else {
        // ctrl
        if (get_mods() & MOD_BIT(KC_LCTRL)) {
            if (clockwise) {
                tap_code(KC_WH_R);
            } else {
                tap_code(KC_WH_L);
            }
        } else {
            if (clockwise) {
                tap_code(KC_WH_D);
            } else {
                tap_code(KC_WH_U);
            }
        }
    }
 
    return false;
}
#endif
```

Reply:
This is the matching #endif for line 19 (#ifdef RGB_MATRIX_ENABLE). Should that be removed as well? Please provide guidance.

Reply:
This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.


As for the endif, the "removal" is just how git shows the diff.

Reply:
Committed this twice by mistake, inserted code twice. Fixed.